zpay32/invoice_test.go

```diff
@@ -515,10 +530,56 @@ func TestDecodeEncode(t *testing.T) {
 				return i
 			},
 		},
+		{
+			// Decode a litecoin testnet invoice
+			encodedInvoice: "lntltc241pvjluezpp5qqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqypqhp58yjmdan79s6qqdhdzgynm4zwqd5d7xmw5fk98klysy043l2ahrqsnp4q0n326hr8v9zprg8gsvezcch06gfaqqhde2aj730yg0durunfhv66m2eq2fx9uctzkmj30meaghyskkgsd6geap5qg9j2ae444z24a4p8xg3a6g73p8l7d689vtrlgzj0wyx2h6atq8dfty7wmkt4frx9g9sp730h5a",
+			network:        "tltc",
+			valid:          true,
+			decodedInvoice: func() *Invoice {
+				return &Invoice{
+					Net:             &ltcTestNetParams,
+					MilliSat:        &testMillisat24BTC,
+					Timestamp:       time.Unix(1496314658, 0),
+					PaymentHash:     &testPaymentHash,
+					DescriptionHash: &testDescriptionHash,
+					Destination:     testPubKey,
+				}
+			},
+		},
+		{
+			// Decode a litecoin mainnet invoice
+			encodedInvoice: "lnltc241pvjluezpp5qqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqqqsyqcyq5rqwzqfqypqhp58yjmdan79s6qqdhdzgynm4zwqd5d7xmw5fk98klysy043l2ahrqsnp4q0n326hr8v9zprg8gsvezcch06gfaqqhde2aj730yg0durunfhv66859t2d55efrxdlgqg9hdqskfstdmyssdw4fjc8qdl522ct885pqk7acn2aczh0jeht0xhuhnkmm3h0qsrxedlwm9x86787zzn4qwwwcpjkl3t2",
+			network:        "ltc",
+			valid:          true,
+			decodedInvoice: func() *Invoice {
+				return &Invoice{
+					Net:             &ltcMainNetParams,
```

Comment:
hm, why not just use `litecoinCfg.MainNetParams`?

Reply:
Therein lies the need for an interface in a future PR. `ltcd.chaincfg.MainNetParams` is a type mismatch with `btcd.chaincfg.MainNetParams`.

Reply:
True... Can cout add a todo here and where you initialize the params, to remind us to create this interface?